Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,284 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        private const string InValidOutput = "N/A.";
+
+        public static bool IsInValidValidInput(string[] input) => input == null || input.Length == 0;
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var personWithHighestBalance = new StringBuilder();
+            var highestBalanceEver = FindHighestBalance(peopleAndBalances, personWithHighestBalance);
+
+            if (personWithHighestBalance.ToString().Contains(", "))
+            {
+               ReplaceCommaWithAnd(personWithHighestBalance); 
+            }
+
+            return $"{personWithHighestBalance.ToString()} had the most money ever. ¤{highestBalanceEver}.";
+        }
+
+        private static void ReplaceCommaWithAnd(StringBuilder message)
+        {
+            var index = message.ToString().LastIndexOf(", ");
+            message.Remove(index, 2).Insert(index, " and "); 
+        }
+
+        private static decimal FindHighestBalance(string[] peopleAndBalances, StringBuilder PersonWithHighestBalance)
```

Comment:
The second parameter should start with p.

Reply:
changed